docs/reference.md

```diff
@@ -131,3 +131,57 @@ dedicated test-suite, consisting of several tests exercising various runtime
 behaviors, for each StableHLO Op. The tests can be found
 [here](https://github.com/openxla/stablehlo/tree/main/stablehlo/tests/) (e.g.
 interpret\_\*.mlir).
+
+### Testing guidelines
+
+**(G1) Do we need to test for all the supported types for every op ?**
+
+We can use the following thumb rule to decide it:
+
+While implementing an op, if we need to introduce code in the corresponding
+`eval` function to handle types differently, then it is imperative to have tests
+to cover for all those types. But in cases where a set of types are handled
+uniformly, like for `add` op, all the supported integer types are handled alike
+using `llvm::APInt` APIs, then we can skip adding tests for all those types and,
+instead, add a single test with a type, with maximum bit-width (an arbitrary
+choice), from the set.
```

Comment:
Also, what happens to existing tests, e.g. https://github.com/openxla/stablehlo/blob/main/stablehlo/tests/interpret_add.mlir?

Reply:
This is my plan:

As suggested in https://github.com/openxla/stablehlo/pull/816#discussion_r1056874546, we will have a separate file `interpret_types.mlir` for testing how types are handled. We can use any particular op to write those tests. Let's say `add` op.

For the existing  interpret_<op>.mlir files, which  violates the guidelines mentioned in this OR, we need to trim them accordingly. As as example, https://github.com/openxla/stablehlo/blob/main/stablehlo/tests/interpret_add.mlir will have fewer tests. Same with other , and other similar files. 

I wish to include the new interpret_types.mlir plus  the trimmed versions of existing test files all in this PR only once we LGTM on the rules.

Reply:
Sounds like a plan. Instead of updating the tests right in this PR, consider opening a follow-up ticket for that, so that we can expedite merging this PR.

Reply:
Here is the [ticket](https://github.com/openxla/stablehlo/issues/918).